src/ucp/core/ucp_ep.c

```diff
@@ -804,10 +804,31 @@ static unsigned ucp_ep_do_disconnect(void *arg)
     return 0;
 }
 
+static void ucp_ep_set_close_request(ucp_ep_h ep, ucp_request_t *request)
+{
+    ucp_ep_flush_state_invalidate(ep);
+    ucp_ep_ext_gen(ep)->close_req.req = request;
+    ep->flags                        |= UCP_EP_FLAG_CLOSE_REQ_VALID;
+}
+
 static void ucp_ep_close_flushed_callback(ucp_request_t *req)
 {
     ucp_ep_h ep = req->send.ep;
 
+    if (ucp_ep_get_cm_lane(ep) != UCP_NULL_LANE) {
+        UCS_ASYNC_BLOCK(&ep->worker->async);
```

Comment:
why is checking cm lane outside ASYNC_BLOCK?
it seems there is common 'if' with line 865-866:
ucp_ep_cm_is_connected() { return cm_lane!=NULL && ep->flags & LOCAL_CONNECTED; }